### PR TITLE
Infrastructure: Upgrade to Typescript 2

### PR DESCRIPTION
**Pull request by CalvinFernandez**
Upgrades to typescript 2.0.6

### CODE REVIEW

**Comment by giladgray** [comment]
👍  change LGTM

File: test/testMethods.ts [hunk 1]
```diff
@@ -323,7 +323,7 @@ namespace TestMethods {
   export function triggerFakeKeyboardEvent(type: string, target: d3.Selection<void>, keyCode: number, options?: {[key: string]: any}) {
     let event = <KeyboardEvent> document.createEvent("Events");
     event.initEvent(type, true, true);
-    event.keyCode = keyCode;
+    (event as any).keyCode = keyCode;
```
Comment (giladgray) [inline test/testMethods.ts:326]: i know we've had issues with creating native events (particularly for Phantom), @themadcreator may be able to provide some more context. though if this works then we're good to go!

**Comment by jkillian** [comment]
looks fine to me as well 👍

File: test/testMethods.ts (hunk 1 above)
Comment (CalvinFernandez) [reply test/testMethods.ts:326]: ya this felt wrong as these are all readonly, but test passes so ...

**Pull request merged**